prow/sidecar/run.go

```diff
@@ -101,33 +112,29 @@ func (o Options) Run(ctx context.Context) (int, error) {
 	// uploading, so we ignore the signals.
 	signal.Ignore(os.Interrupt, syscall.SIGTERM)
 
-	buildLog := logReader(entries)
+	buildLogs := logReaders(entries)
 	metadata := combineMetadata(entries)
-	return failures, o.doUpload(spec, passed, aborted, metadata, buildLog)
+	return failures, o.doUpload(spec, passed, aborted, metadata, buildLogs)
 }
 
 const errorKey = "sidecar-errors"
 
-func start(part string) string {
-	return fmt.Sprintf("\n==== start of %s log ====\n", part)
-}
-
-func logReader(entries []wrapper.Options) io.Reader {
-	var readers []io.Reader
-	for i, opt := range entries {
-		ent := nameEntry(i, opt)
+func logReaders(entries []wrapper.Options) map[string]io.Reader {
+	readers := make(map[string]io.Reader)
+	for _, opt := range entries {
+		log, err := os.Open(opt.ProcessLog)
+		buildLog := "build-log.txt"
 		if len(entries) > 1 {
-			readers = append(readers, strings.NewReader(start(ent)))
+			buildLog = fmt.Sprintf("%s-build-log.txt", strings.TrimSuffix(filepath.Base(opt.ProcessLog), "-log.txt"))
```

Comment:
Yeah I agree this would be much cleaner.  I will also need to make some very small changes to the podspec decoration to make sure to pass the container name in the wrapper options.